modules/fedimint-lnv2-tests/src/bin/self_payment_with_two_gateways.rs

```diff
@@ -0,0 +1,84 @@
+use devimint::version_constants::VERSION_0_4_0_ALPHA;
+use devimint::{cmd, util};
+use fedimint_core::util::SafeUrl;
+use tracing::info;
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    devimint::run_devfed_test(|dev_fed| async move {
+        let fedimint_cli_version = util::FedimintCli::version_or_default().await;
+        let fedimintd_version = util::FedimintdCmd::version_or_default().await;
+        let gatewayd_version = util::Gatewayd::version_or_default().await;
+
+        if fedimint_cli_version < *VERSION_0_4_0_ALPHA {
+            info!(%fedimint_cli_version, "Version did not support lnv2 module, skipping");
+            return Ok(());
+        }
+
+        if fedimintd_version < *VERSION_0_4_0_ALPHA {
+            info!(%fedimintd_version, "Version did not support lnv2 module, skipping");
+            return Ok(());
+        }
+
+        if gatewayd_version < *VERSION_0_4_0_ALPHA {
+            info!(%gatewayd_version, "Version did not support lnv2 module, skipping");
+            return Ok(());
+        }
+
+        let client = dev_fed
+            .fed()
+            .await?
+            .new_joined_client("lnv2-module-client")
+            .await?;
+
+        let gw_lnd = devfed.gw_lnd().addr;
+        let gw_cln = devfed.gw_cln().addr;
+
+        let result: Result<(Bolt11Invoice, OperationId), FetchInvoiceError> =
+            serde_json::from_value(
+                cmd!(client, "module", "lnv2", "receive", gw_lnd, "100000")
+                    .out_json()
+                    .await?,
+            );
+
+        let Ok(invoice, receive_op) = result?;
```

Comment:
Splitting these results into separate vars / operations is a bit ... odd. Is there a reason for it?

`serde_json::from_value::<(Bolt11Invoice, OperationId)>::(/*...*/)` would be more idiomatic, but then I noticed that you specify `FetchInvoiceError` and I wonder if it is even correct. It possibly might be working only because the error type is actually never there. `from_value` return deserialization result ...

I'm confused myself now ... ;D

Reply:
Fixed